src/System.Net.Mail/tests/Functional/SmtpClientTest.cs

```diff
@@ -305,25 +305,31 @@ public void TestMailDelivery()
             }
         }
 
-        [Fact]
-        public async Task TestMailDeliveryAsync()
+        [Theory]
+        [InlineData("howdydoo")]
+        [InlineData("")]
+        [InlineData(null)]
+        public void TestMailDeliveryAsync(string body)
         {
             SmtpServer server = new SmtpServer();
             SmtpClient client = new SmtpClient("localhost", server.EndPoint.Port);
-            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", "howdydoo");
+            MailMessage msg = new MailMessage("foo@example.com", "bar@example.com", "hello", body);
             string clientDomain = IPGlobalProperties.GetIPGlobalProperties().HostName.Trim().ToLower();
 
             try
             {
                 Thread t = new Thread(server.Run);
                 t.Start();
-                await client.SendMailAsync(msg);
+                using (var cts = new CancellationTokenSource(TimeSpan.FromSeconds(30)))
+                {
+                    client.SendMailAsync(msg).Wait(cts.Token);
```

Comment:
I preferred use cts+wait intead of `Task.Delay` for code clarity, let me know if it's ok

Reply:
Using any form of Wait() will block the thread.  We always use some await pattern instead.

Why are you using any "timeout" pattern at all?  Is it because if the test fails (i.e. your fix isn't present), then the test will hang?  If so, you should use the current Task extensions with timeout patterns we already have in other tests.

Reply:
Will do